core/core.go

```diff
@@ -1,40 +1,39 @@
 package core
 
 import (
+	"os"
+	"os/signal"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/xh3b4sd/anna/common"
 	"github.com/xh3b4sd/anna/gateway"
 	gatewayspec "github.com/xh3b4sd/anna/gateway/spec"
-	"github.com/xh3b4sd/anna/impulse"
 	"github.com/xh3b4sd/anna/log"
 	"github.com/xh3b4sd/anna/spec"
 	"github.com/xh3b4sd/anna/state"
 )
 
 type Config struct {
-	TextGateway gatewayspec.Gateway `json:"-"`
+	FactoryClient spec.Factory `json:"-"`
```

Comment:
It is probably because the other objects still referenced to their own gateway during manual testing.